index.html

```diff
@@ -1294,39 +1294,105 @@ <h2>DID Subject</h2>
+of five cryptographic keys from which any three are required to contribute to
+a threshold signature. Methods need not be cryptographic. An example of this
+might be the contact information for a biometric service provider that compares
+a purported <a>DID controller</a> against a candidate biometric vector.
+      </p>
 
       <p>
-A <a>DID document</a> can express cryptographic keys and other verification
-methods, which can be used to authenticate or authorize interactions with the
-<a>DID subject</a> or associated parties. The information expressed often
-includes globally unambiguous identifiers and public key material, which can be
-used to verify digital signatures. Other information can be expressed, such as
-attributes that enable one to determine whether it is a
-hardware-backed cryptographic key.
+A <a>DID document</a> MAY include a <code>verificationMethod</code> property.
       </p>
 
+      <dl>
+        <dt><dfn>verificationMethod</dfn></dt>
+        <dd>
+If a <a>DID document</a> includes a <code>verificationMethod</code> property,
+the value of the property MUST be an array of verification method objects.
+Each verification method object MUST have the <code>id</code>, <code>type</code>,
+<code>controller</code>, and specific verification method properties. The
+verification method objects MAY include additional properties.
+        </dd>
+      </dl>
+
       <p>
-Regarding cryptographic key material, public keys can be included in a
-<a>DID document</a> using, for example, the <code>publicKey</code> or
-<code>authentication</code> properties, depending on what they are to be used
-for. Each public key has an identifier (<code>id</code>) of its own, a
-<code>type</code>, and a <code>controller</code>, as well as other properties
-that depend on the type of key it is.
+The value of the <code>id</code> property MUST be a <a>URI</a>. The array of
+verification methods MUST NOT contain multiple entries with the same
+<code>id</code>. In this case, a <a>DID document</a> processor MUST produce an
+error.
       </p>
 
       <p>
-This specification strives to limit the number of formats for expressing public
-key material in a DID Document to the fewest possible, to increase the
-likelihood of interoperability. The fewer formats that implementers have to
-implement, the more likely it will be that they will support all of them. This
-approach attempts to strike a delicate balance between ease of implementation
-and supporting formats that have historically had broad deployment. The specific
-types of key formats that are supported in this specification are listed below.
+The value of the <code>type</code> property MUST be exactly one verification
+method type. The verification method type SHOULD be registered in the
+[[DID-SPEC-REGISTRIES]] for maximum interoperability.
+      </p>
+
+      <p>
+The value of the <code>controller</code> property, MUST be a valid <a>DID</a>.
```

Comment:
Hmm... wonder if we want to be this prescriptive or just say "MUST be a valid URI". Thinking of did:web here... I'm torn, can see arguments both ways.

Reply:
This is just what it says for publicKey. Perhaps it warrants an issue of its own? I may be mistaken, but I thought it still ought to be a DID for did:web (it's just that the DID would have a http uri as part of it)